### PR TITLE
Refactors address_of to use Go 1.18 generics

### PR DESCRIPTION
**Pull request by toszr**
# Description

Refactors `address_of` package to make use of the brand new Go 1.18 generics (see [official tutorial](https://go.dev/doc/tutorial/generics)).

## How can this be tested?
`make lint && make test`

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly

### CODE REVIEW

**Review by mjgrzybek — APPROVED** [review]

**Review by mjgrzybek — CHANGES_REQUESTED** [review]

File: src/controllers/dynakube/oneagent/daemonset/daemonset.go [hunk 1]
```diff
@@ -212,7 +212,7 @@ func (dsInfo *builderInfo) podSpec() corev1.PodSpec {
 		DNSPolicy:                     dnsPolicy,
 		Volumes:                       volumes,
 		Affinity:                      affinity,
-		TerminationGracePeriodSeconds: address_of.Int64(defaultTerminationGracePeriod),
+		TerminationGracePeriodSeconds: address.Of[int64](defaultTerminationGracePeriod),
```
[the remark this answers is not in the document]
Comment (toszr) [reply src/controllers/dynakube/oneagent/daemonset/daemonset.go:215]: Actually, no - type inference won't work for pointers:
image

[the remark this answers is not in the document]
Comment (mjgrzybek) [reply src/controllers/dynakube/oneagent/daemonset/daemonset.go:215]: `defaultTerminationGracePeriod` is an `int`.
`address.Of[int64]` copies argument to `int64` parameter and return pointer to it. 
It works but cleaner solution is to change `defaultTerminationGracePeriod` type to `int64` and apply suggestion above.

**Review by mjgrzybek — APPROVED** [review]

File: src/kubeobjects/address/address_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOf(t *testing.T) {
+	t.Run("bool", func(t *testing.T) {
+		assert.True(t, *Of(true))
+		assert.False(t, *Of(false))
+
+		const definitelyYes = true
```
Comment (meik99) [inline src/kubeobjects/address/address_test.go:14]:
```suggestion
		const constantBool = true
```

File: src/kubeobjects/address/address_test.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		assert.True(t, *Of(probablyYes))
+
+		probablyYes = false
+		assert.False(t, *Of(probablyYes))
+	})
+
+	t.Run("int64", func(t *testing.T) {
+		assert.Equal(t, 23, *Of(23))
+
+		const twentySeven = 27
```
Comment (meik99) [inline src/kubeobjects/address/address_test.go:27]:
```suggestion
		const constantInt = 27
```

File: src/kubeobjects/address/address_test.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+		assert.Equal(t, 27, *Of(twentySeven))
+
+		roughlyThree := int64(4)
```
Comment (meik99) [inline src/kubeobjects/address/address_test.go:30]:
```suggestion
		mutableInt := int64(4)
```

**Review by meik99 — CHANGES_REQUESTED** [review]

**Review by mjgrzybek — APPROVED** [review]

**Review by luhi-DT — APPROVED** [review]

**Pull request merged**